pmlc/dialect/pxa/transforms/tile.cc

```diff
@@ -31,11 +32,23 @@ void performTiling(AffineParallelOp op, llvm::ArrayRef<int64_t> tileSizes) {
   // Make the inner parallel for (abve all other code);
   auto inner = builder.create<AffineParallelOp>(
       op.getLoc(), op.getResultTypes(), lbMap, outerIdxs, ubMap, outerIdxs);
-  // Splice instructions into the interior
+  // Build a map from old indices to new indices
+  mlir::BlockAndValueMapping mapper;
+  auto innerIdxs = inner.getBody()->getArguments();
+  for (unsigned i = 0; i < outerIdxs.size(); ++i) {
+    mapper.map(outerIdxs[i], innerIdxs[i]);
+  }
+  // Clone new instructions into the interior
   auto &innerLoopOps = inner.getBody()->getOperations();
   auto &outerLoopOps = outerBody->getOperations();
-  innerLoopOps.splice(std::prev(innerLoopOps.end()), outerLoopOps,
-                      std::next(outerLoopOps.begin(), 1), outerLoopOps.end());
+  for (auto& origOp : outerLoopOps) {
+    if (&origOp == inner.getOperation()) {
+      continue;
+    }
+    Operation* newOp = origOp.clone(mapper);
```

Comment:
This does a deep copy, can we instead 1) splice the interior in 2) walk the uses of the outer args and replace them with the inners args (except the uses in the actual inner PF begin/end).  Or alternately 1) Make the inner AP with fake ranges (to get values for the inner args constructed), 2) splice the interior in 3) For each dim, call outerArg.replaceAllUsesWith(innerArg) 4) Update the inner AP to have ranges defined by the outer args.  This is a pattern I have seen before in MLIR code.  Either of these makes the big-O for the transform be the number of uses of the index rather than the entire interior of the loop.  

It's probably not critical for performance yet, and the code currently is readable and correct, so if you would rather me just  OK this PR, we can always get to this if/when we run into perf issues, if so just let me know.